doc/headset.md

```diff
@@ -0,0 +1,430 @@
+
+Declaration:
+``` ts
+    sdk.headset.headsetEvents.subscribe(event: {
+        event: 'deviceAnsweredCall',
+        payload: {
+            name: string,
+            code?: string,
+            event: { `containing various items mostly for logging purposes` }
+        }
+    } => {
+        if (event.event === 'deviceAnsweredCall') {
+            sdk.acceptPendingSession();
+        }
+    })
+```
+
+Value of event:
+* `event: EmittedHeadsetEvents` - string value emitted by the headset library to determine what event had just occurred
+* `payload:` {
+    `name: string` - Name of the recent event as interpretted by the headset device
+    `event`: { containing various items mostly for logging purposes}
+    `code?: string` - Optional: A string value of a number that represents the action that was just taken. Not all vendors supply a code which is why it is only optional
+}
+
+
+#### `deviceEndedCall`
+Event emitted when a user presses the answer call button while in an active call on their selected device. The event includes the event `name` as it is interpretted by the headset and a collection of items that may help with logging (`event`). It can also potentially have a `code` that corresponds to the event.
+
+Declaration:
+``` ts
+    sdk.headset.headsetEvents.subscribe(event: {
+        event: 'deviceEndedCall',
+        payload: {
+            name: string,
+            event: { `containing various items mostly for logging purposes` },
+            code?: string
+        } => {
+            if (event.event === 'deviceEndedCall) {
+                sdk.endSession();
+            }
+        }
+    })
+```
+Value of event:
+* `event: EmittedHeadsetEvents` - string value emitted by the headset library to determine what event had just occurred
+* `payload:` {
+    `name: string` - Name of the recent event as interpretted by the headset device
+    `event`: { containing various items mostly for logging purposes}
+    `code?: string` - Optional: A string value of a number that represents the action that was just taken. Not all vendors supply a code which is why it is only optional
+}
+
+
+#### `deviceMuteStatusChanged`
+Event emitted when a user presses the mute call button on their selected device. It doesn't matter if the device state is currently muted or unmuted, this event will be emitted with the _OPPOSITE_ value. For example, if the headset is currently muted, it will emit the event with the corresponding value to unmute the device. The event includes the event `name` as it is interpretted by the headset and a collection of items that may help with logging (`event`). It also comes with a value known as `isMuted` which determines the event is trying to mute or unmute the call. It can also potentially have a `code` that corresponds to the event.
```

Comment:
When this says _OPPOSITE_, does that mean if `isMuted === true` and then I make a duplicate call to `setMute(true)` – will this emit `isMuted === false` just because it is opposite of what I called `setMute` with? Again, this may just be a lack of knowledge on my part.

Reply:
I think I more so wrote it from a standpoint of using the headset itself rather than a coding perspective.  Like if the mute call button is currently lit up on the device, pressing it again will emit `isMuted === false`. The opposite is also true where if the light on the device is off but a user presses it, it will then emit `isMuted === true`.  If however you make a duplicate function call that explicitly passes in true while `isMuted === true`, then `isMuted` should still remain true